earn/src/components/lend/modal/content/ToUniswapNFTModalContent.tsx

```diff
@@ -0,0 +1,199 @@
+      [0, 0],
+    ],
+    enabled: Boolean(userAddress),
+    chainId: activeChain.id,
+  });
+  const gasLimit = withdrawConfig.request?.gasLimit.mul(GAS_ESTIMATE_WIGGLE_ROOM).div(100);
+  const { write: contractWrite, isLoading: isAskingUserToConfirm } = useContractWrite({
+    ...withdrawConfig,
+    request: {
+      ...withdrawConfig.request,
+      gasLimit,
+    },
+    onSuccess(data) {
+      setIsOpen(false);
+      setPendingTxnResult(data);
+    },
+  });
+
+  let confirmButtonState = ConfirmButtonState.READY;
+  if (withdrawError !== null) {
+    confirmButtonState = ConfirmButtonState.CONTRACT_ERROR;
+  } else if (contractWrite === undefined || !withdrawConfig.request) {
+    confirmButtonState = ConfirmButtonState.LOADING;
+  } else if (isAskingUserToConfirm) {
+    confirmButtonState = ConfirmButtonState.WAITING_FOR_USER;
+  }
+
+  const confirmButton = getConfirmButton(confirmButtonState);
+
+  return (
+    <FilledStylizedButton size='M' fillWidth={true} disabled={!confirmButton.enabled} onClick={contractWrite}>
+      {confirmButton.text}
+    </FilledStylizedButton>
+  );
+}
+
+export type RemoveCollateralModalContentProps = {
+  borrower: BorrowerNftBorrower;
+  positionToWithdraw: UniswapPosition;
+  uniswapNftId: BigNumber;
+  setIsOpen: (isOpen: boolean) => void;
+  setPendingTxnResult: (result: SendTransactionResult | null) => void;
+};
+
+export default function ToUniswapNFTModalContent(props: RemoveCollateralModalContentProps) {
+  const { borrower, positionToWithdraw, uniswapNftId, setIsOpen, setPendingTxnResult } = props;
+
+  const { health: newHealth } = isHealthy(
+    borrower.assets,
+    borrower.liabilities,
+    [],
+    borrower.sqrtPriceX96,
+    borrower.iv,
+    borrower.nSigma,
+    borrower.token0.decimals,
+    borrower.token1.decimals
+  );
+
+  const canExecute = newHealth > 1.0;
+
+  // TODO: could provide links (in summary) to both the BorrowerNFT and UniswapNFT on OpenSea
+  return (
+    <>
+      <div className='flex flex-col gap-1 w-full'>
+        <HealthBar health={newHealth} />
+        <Text size='M' weight='bold' className='mt-4'>
+          Summary
+        </Text>
+        {canExecute ? (
+          <Text size='XS' color={SECONDARY_COLOR} className='overflow-hidden text-ellipsis'>
+            You have an {borrower.token0.symbol}/{borrower.token1.symbol} Uniswap Position in the range{' '}
+            {positionToWithdraw.lower}&nbsp;⇔&nbsp;{positionToWithdraw.upper}. You're moving it from an Aloe Borrower
+            NFT back to a plain Uniswap NFT (#{uniswapNftId.toString()}). It will no longer act as collateral.
+          </Text>
+        ) : (
+          <Text size='XS' color={SECONDARY_COLOR} className='overflow-hidden text-ellipsis'>
+            This Uniswap NFT is the only thing keeping your position healthy. Before you can withdraw, you must repay
+            some (or all) borrows.
+          </Text>
+        )}
+      </div>
+      {canExecute && (
```

Comment:
Why not just disable the button if we can't execute rather than not show it?

Reply:
Because loading that component would result in bogus (failing) transaction gas estimations and unnecessary Alchemy calls.